iOS/G3MiOSSDK/Commons/Basic/PyramidElevationDataProvider.cpp

```diff
@@ -0,0 +1,240 @@
+//
+//  PyramidElevationDataProvider.cpp
+//  G3MiOSSDK
+//
+//  Created by Sebastian Ortega Trujillo on 4/3/16.
+//
+//
+
+#include "PyramidElevationDataProvider.hpp"
+#include "ShortBufferElevationData.hpp"
+#include "IStringBuilder.hpp"
+#include "DownloadPriority.hpp"
+#include "G3MContext.hpp"
+#include "TimeInterval.hpp"
+#include <sstream>
+
+
+class PyramidElevationDataProvider_BufferDownloadListener : public IBufferDownloadListener {
+private:
+    
+    const Sector &_sector;
+    int _width, _height;
+    IElevationDataListener *_listener;
+    bool _autodeleteListener;
+    double _deltaHeight;
+    
+#ifdef C_CODE
+    const Vector2I* getResolution(const JSONObject *data){
+        return new Vector2I((int) data->getAsNumber("width",0),(int) data->getAsNumber("height",0));
+    }
+#endif
+#ifdef JAVA_CODE
+    private Vector2I getResolution(JSONObject data){
+        return new Vector2I((int) data.getAsNumber("width",0),(int) data.getAsNumber("height",0));
+    }
+#endif
+    
+    ShortBufferElevationData* getElevationData(const Sector& sector,
+                                               const Vector2I& extent,
+                                               const JSONObject *data,
+                                               double deltaHeight){
+        const short minValue = IMathUtils::instance()->minInt16();
+        const int size = extent._x * extent._y;
+        const JSONArray *dataArray = data->getAsArray("data");
+        short *shortBuffer = new short[size];
+        for (int i = 0; i < size; i++)
+        {
+            short height = (short) dataArray->getAsNumber(i, minValue);
+            
+            if (height == 15000) //Our own NODATA, since -9999 is a valid height.
+            {
```

Comment:
I think that the constant 15000 has to be a parameter to the PyramidElevationDataProvider constructor.